eclair-node/src/main/scala/fr/acinq/eclair/api/directives/RouteFormat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.api.directives
+
+import fr.acinq.eclair.router.Router.RouteResponse
+
+// @formatter:off
+sealed trait RouteFormat
+case object NodeIdRouteFormat extends RouteFormat
+case object ShortChannelIdRouteFormat extends RouteFormat
+// @formatter:on
+
+object RouteFormat {
+
+  val NODE_ID = "nodeId"
+  val SHORT_CHANNEL_ID = "shortChannelId"
+
+  def fromString(s: String): RouteFormat = s match {
+    case NODE_ID => NodeIdRouteFormat
+    case SHORT_CHANNEL_ID => ShortChannelIdRouteFormat
+    case _ => throw new IllegalArgumentException(s"invalid route format, possible values are ($NODE_ID, $SHORT_CHANNEL_ID)")
```

Comment:
Can we avoid throwing here and return an `Option` or an `Either` instead?

Reply:
Bit it throws here, for example: https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/ShortChannelId.scala#L44

There will to much boilerplate to handle `Option` or `Either` here.

Reply:
Ok fine, let's skip it for now, we'll refactor the whole API to avoid throwing exceptions in a separate PR.

Reply:
This is how it looks like in the CLI:
```bash
$ ./eclair-cli findroutetonode --format= --nodeId=0321fb8a1edde1e0d1721e726989b5c28264d56796ba968a4529c19f05cbf63746 --amountMsat=1000 
The form field 'format' was malformed:
invalid route format, possible values are (nodeId, shortChannelId)
```